src/sentry/tasks/deletion/hybrid_cloud.py

```diff
@@ -0,0 +1,214 @@
+        v = client.get(key)
+        if v is None:
+            result = (0, uuid4().hex)
+            client.set(key, json.dumps(result))
+            return result
+        return tuple(json.loads(v))
+
+
+def set_watermark(
+    prefix: str, field: HybridCloudForeignKey, value: int, prev_transaction_id: str
+) -> None:
+    with redis.clusters.get("default").get_local_client_for_key("deletions.watermark") as client:
+        client.set(
+            get_watermark_key(prefix, field),
+            json.dumps((value, sha1(prev_transaction_id.encode("utf8")).hexdigest())),
+        )
+    metrics.gauge(
+        "deletion.hybrid_cloud.low_bound",
+        value,
+        tags=dict(
+            table_name=field.model._meta.db_table,
+            field_name=field.name,
+            watermark=prefix,
+        ),
+    )
+
+
+def chunk_watermark_batch(
+    prefix: str, field: HybridCloudForeignKey, manager: Manager, *, batch_size: int
+) -> Tuple[int, int, bool, str]:
+    lower, transaction_id = get_watermark(prefix, field)
+    upper = manager.aggregate(Max("id"))["id__max"] or 0
+    batch_upper = min(upper, lower + batch_size)
+    return lower, upper, batch_upper + batch_size < upper, transaction_id
+
+
+@instrumented_task(
+    name="sentry.tasks.deletion.hybrid_cloud.schedule_hybrid_cloud_foreign_key_jobs",
+    queue="cleanup",
+    acks_late=True,
+)
+def schedule_hybrid_cloud_foreign_key_jobs():
+    for silo_mode in deletion_silo_modes():
+        for app, app_models in apps.all_models.items():
+            for model in app_models.values():
+                if not hasattr(model._meta, "silo_limit"):
+                    continue
+
+                # Only process models local this operational mode.
+                if silo_mode not in model._meta.silo_limit.modes:
+                    continue
+
+                for field in model._meta.fields:
+                    if not isinstance(field, HybridCloudForeignKey):
+                        continue
+
+                    process_hybrid_cloud_foreign_key_cascade_batch.delay(
+                        app_name=app,
+                        model_name=model.__name__,
+                        field_name=field.name,
+                        silo_mode=silo_mode.name,
+                    )
+
+
+@instrumented_task(
+    name="sentry.tasks.deletion.process_hybrid_cloud_foreign_key_cascade_batch",
+    queue="cleanup",
+    acks_late=True,
+)
+def process_hybrid_cloud_foreign_key_cascade_batch(
+    app_name: str, model_name: str, field_name: str, silo_mode: str
+) -> None:
+    try:
+        model = apps.get_model(app_label=app_name, model_name=model_name)
+        field: HybridCloudForeignKey
+        for field in model._meta.fields:
+            if not isinstance(field, HybridCloudForeignKey):
+                continue
+            if field.name == field_name:
+                break
+        else:
+            raise LookupError(f"Could not find field {field_name} on model {app_name}.{model_name}")
```

Comment:
If you know the field name you could use `model._meta.get_field()` to read the field. This method will also raise if the field is bad.

Reply:
interesting, I'll give that a shot.